### DISCUSSION TITLE
How to define Fungible asset transfer Event?

### DISCUSSION THREAD

**Discussion by junyoungcross** [aptos-labs/aptos-developer-discussions#545] [Questions] [answered]
### Describe your question in detail.

I wonder how can i extract transfer in fungible asset event such as `**1657038910**` version

in this version, there are 3 deposit event and 1 withdraw event

it means there are 1 transfer event and 2 mint event

if` event index 2 `is deposit and `event index 4` is withdraw and `event index 5` is deposit.
is` event index 2` to` event index 4` is transfer? or `event index 4 `to `event index 5` is transfer?

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

N/A

**Comment by ying-w** [comment, marked as answer]
For FA use the `store` to determine if it's a (complete) transfer or not.

Events are emitted by the FA/Coin store and done in a double entry accounting kind of way (have one event for withdraw and one event for deposit), index 0 and 1 are coin movements. If a coin isn't moved into an store then there will be no event.

My understanding of what events represent.

0. Withdraw `0x4e1854f6d332c9525e258fb6e66f84b6af8aba687bbcb832a24768c4e175feec::abtc::ABTC` coin from `0x93a63588720871f6e1c20218ce84f8689c0db0d20bb8910b6cf4c48b16797903` storage owned by `0x3535c9b6b97d6c9c5528592ca9db3dc124b11cc1e82023e8b02bc382d03d6e65`
1. Deposit `0x4e1854f6d332c9525e258fb6e66f84b6af8aba687bbcb832a24768c4e175feec::abtc::ABTC` coin to `0x03e90428a4d635bd398dce9b93fd6570f8219ee47f331041f76c7927c4d8468a` storage owned by `0x07c8e9b0f2ec5621dc150afbd5cb6cb9d0f61ca91921041b5fdfe063e524b851`
2. Deposit `0x2760ca0d7ed74fec37f1c539fd752b5c684fa011c539a79def7a6c67717ebace` (FA migrated ABTC) FA to `0x04e30151a32418243b674252f69d2bcd766ebeb8e34994ebe31e852ee5f40456` storage owned by `0x3535c9b6b97d6c9c5528592ca9db3dc124b11cc1e82023e8b02bc382d03d6e65` - prob FA migration, withdraw of coin is inferred
3. some echo lending event
4. Withdraw `0x2760ca0d7ed74fec37f1c539fd752b5c684fa011c539a79def7a6c67717ebace` (FA migrated ABTC) FA from `0x04e30151a32418243b674252f69d2bcd766ebeb8e34994ebe31e852ee5f40456` storage owned by `0x3535c9b6b97d6c9c5528592ca9db3dc124b11cc1e82023e8b02bc382d03d6e65`
5. Deposit `0x2760ca0d7ed74fec37f1c539fd752b5c684fa011c539a79def7a6c67717ebace` (FA migrated ABTC) FA to `0x410dd94d454c5e3f0a202efb27647eae6c355d5a3fe24f3944e8c8244f7ceeeb` storage owned by `0xfab1043eef2567172318567d1fe40dc7bdf025c0235c941062fd3c3bff03cab2`
6. Freeze `0xee2a8176e27b75bd30419b0ded6c00f15ec551135a6ea69f2575601ec87439a6` FA in `0x99c172717cc252f9ae95bf307d1fd447801cb763e26537586d6937abffc9ed99` storage owned by `0x3535c9b6b97d6c9c5528592ca9db3dc124b11cc1e82023e8b02bc382d03d6e65`
7. Deposit `0xee2a8176e27b75bd30419b0ded6c00f15ec551135a6ea69f2575601ec87439a6` FA in `0x99c172717cc252f9ae95bf307d1fd447801cb763e26537586d6937abffc9ed99` owned by `0x3535c9b6b97d6c9c5528592ca9db3dc124b11cc1e82023e8b02bc382d03d6e65`

Details from graphql (querying pre-parsed tables) https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Findexer.mainnet.aptoslabs.com%2Fv1%2Fgraphql&query=query+MyQuery+%7B%0A++fungible_asset_activities%28%0A++++where%3A+%7Btransaction_version%3A+%7B_eq%3A+%221657038910%22%7D%7D%0A++++order_by%3A+%7Bevent_index%3A+asc%7D%0A++%29+%7B%0A++++event_index%0A++++asset_type%0A++++is_frozen%0A++++owner_address%0A++++storage_id%0A++++token_standard%0A++++type%0A++%7D%0A%7D%0A

**Reply by junyoungcross** [reply]
Thank you for your reply
